libs/sdk-ui-dashboard/src/presentation/dashboard/Dashboard.tsx

```diff
@@ -178,56 +188,60 @@ export const Dashboard: React.FC<IDashboardProps> = (props: IDashboardProps) =>
-            onStateChange={props.onStateChange}
-            onEventingInitialized={props.onEventingInitialized}
-            additionalReduxContext={props.additionalReduxContext}
-            customizationFns={props.customizationFns}
-        >
-            <ToastMessageContextProvider>
-                <ExportDialogContextProvider>
-                    <DashboardCustomizationsProvider
-                        insightMenuItemsProvider={props.insightMenuItemsProvider}
-                    >
-                        <DashboardComponentsProvider
-                            ErrorComponent={props.ErrorComponent ?? DefaultError}
-                            LoadingComponent={props.LoadingComponent ?? DefaultLoading}
-                            LayoutComponent={props.LayoutComponent ?? DefaultDashboardLayout}
-                            InsightComponentProvider={insightProvider}
-                            InsightMenuButtonComponentProvider={insightMenuButtonProvider}
-                            InsightMenuComponentProvider={insightMenuProvider}
-                            KpiComponentProvider={kpiProvider}
-                            WidgetComponentProvider={widgetProvider}
-                            ButtonBarComponent={props.ButtonBarComponent ?? DefaultButtonBar}
-                            MenuButtonComponent={props.MenuButtonComponent ?? DefaultMenuButton}
-                            TopBarComponent={props.TopBarComponent ?? DefaultTopBar}
-                            TitleComponent={props.TitleComponent ?? DefaultTitle}
-                            ScheduledEmailDialogComponent={
-                                props.ScheduledEmailDialogComponent ?? DefaultScheduledEmailDialog
-                            }
-                            ShareDialogComponent={props.ShareDialogComponent ?? DefaultShareDialog}
-                            SaveAsDialogComponent={props.SaveAsDialogComponent ?? DefaultSaveAsDialog}
-                            DashboardAttributeFilterComponentProvider={attributeFilterProvider}
-                            DashboardDateFilterComponent={
-                                props.DashboardDateFilterComponent ?? DefaultDashboardDateFilter
-                            }
-                            FilterBarComponent={props.FilterBarComponent ?? DefaultFilterBar}
-                        >
-                            <DashboardConfigProvider menuButtonConfig={props.menuButtonConfig}>
-                                <DashboardLoading {...props} />
-                            </DashboardConfigProvider>
-                        </DashboardComponentsProvider>
-                    </DashboardCustomizationsProvider>
-                </ExportDialogContextProvider>
-            </ToastMessageContextProvider>
-        </DashboardStoreProvider>
+        <BackendProvider backend={backend}>
+            <WorkspaceProvider workspace={workspace}>
+                <DashboardStoreProvider
```

Comment:
what's up with these? why are they needed suddenly?

Reply:
Hi this is base on discussion with Dan ... we allow pass workspace and backend as a props in Dashboard but then we are force to pass this props to internal component like DashboardHeader to ShareDialogDashboardHeader